photon-ml/src/test/scala/com/linkedin/photon/ml/data/LabeledPointTest.scala

```diff
@@ -89,10 +96,56 @@ class LabeledPointTest {
   @DataProvider(name = "dataProvider")
   def dataProvider(): Array[Array[Any]] = {
     Array(
+      Array(DenseVector(1.0, 0.0, 0.4, 0.5), 12.34, DenseVector(0.0, 0.0, 0.0, 0.0), 12.34),
       Array(DenseVector(1.0, 0.0, 0.4, 0.5), 1.0, DenseVector(-1.0, -0.5, 0.1, 0.0), 0.04),
       Array(SparseVector(4)((0, 1.0), (2, 0.4), (3, 0.5)), -1.0, DenseVector(-1.0, -0.5, 0.1, 0.0), -1.96),
       Array(DenseVector(1.0, 0.0, 0.4, 0.5), 0.0, SparseVector(4)((0, -1.0), (1, -0.5), (2, 0.1)), -0.96),
       Array(SparseVector(4)((0, 1.0), (2, 0.4), (3, 0.5)), -100.0, SparseVector(4)((0, -1.0), (1, -0.5), (2, 0.1)), -100.96)
     )
   }
+
+  /**
+    * Test computeMargin by comparing to the explicit form of calculation on random samples
+    *
+    * @param datum Randomly generated labeled samples
+    */
+  @Test(dataProvider = "getRandomDataPoints")
+  def testComputeMarginOnRandomlyGeneratedPoints(datum: LabeledPoint): Unit =
+  {
+    // Init random coefficients
+    val r: Random = new Random(LabeledPointTest.PARAMETER_RANDOM_SEED)
+    val coef: Vector[Double] = DenseVector.fill[Double](DataPointTest.PROBLEM_DIMENSION){ r.nextDouble() }
```

Comment:
There's a utility function for this:

https://github.com/linkedin/photon-ml/blob/master/photon-test/src/main/scala/com/linkedin/photon/ml/test/CommonTestUtils.scala#L75